backend/dal/Services/Concrete/NotificationQueueService.cs

```diff
@@ -389,6 +389,44 @@ public NotificationQueue GenerateNotification<T>(string to, NotificationTemplate
             }
             return notification;
         }
+        
+
+        /// <summary>
+        /// Generates the access request notification for the specified 'model' and 'template'.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="template"></param>
+        /// <param name="model"></param>
+        /// <param name="sendOn"></param>
+        /// <returns></returns>
+        public NotificationQueue GenerateNotification<T>(NotificationTemplate template, T model, DateTime? sendOn = null)
+        {
+            if (template == null) throw new ArgumentNullException(nameof(template));
+            if (model == null) throw new ArgumentNullException(nameof(model));
```

Comment:
Add a check to ensure it has a "to".
`if (String.IsNullOrEmpty(template.To)) throw new ArgumentException("Template requires a 'to'.", nameof(template));